src/l500/l500-options.cpp

```diff
@@ -64,27 +93,50 @@ namespace librealsense
         auto min_value = float(*(reinterpret_cast<int32_t*>(min.data())));
 
         _range = option_range{ min_value,
-            max_value,
-            float(*(reinterpret_cast<int32_t*>(step.data()))),
-            def };
+                               max_value,
+                               float( *( reinterpret_cast< int32_t * >( step.data() ) ) ),
+                               query_default() };
+    }
+
+    void l500_hw_options::update_default()
+    {
+        _range.def = query_default();
     }
 
     void l500_hw_options::enable_recording(std::function<void(const option&)> recording_action)
     {}
 
-    float l500_hw_options::query(int mode) const
+    float l500_hw_options::query( l500_command type, int mode ) const
     {
-        auto res = _hw_monitor->send(command{ AMCGET, _type, get_current, mode });
-
-        if (res.size() < sizeof(int32_t))
+         if (type == get_current)
         {
-            std::stringstream s;
-            s << "Size of data returned is not valid min size = " << res.size();
-            throw std::runtime_error(s.str());
+            auto res = _hw_monitor->send( command{ AMCGET, _type, get_current, mode } );
+
+            if( res.size() < sizeof( int32_t ) )
+            {
+                std::stringstream s;
+                s << "Size of data returned is not valid min size = " << res.size();
+                throw std::runtime_error( s.str() );
+            }
+            auto val = *( reinterpret_cast< int32_t * >( (void *)res.data() ) );
+            return float( val );
         }
+        else if (type == get_default)
+        {
+            const int digital_gain_num_of_values = RS2_DIGITAL_GAIN_LOW;
+            auto digital_gain_val = _digital_gain->query();
```

Comment:
But what will happen when the digital gain is on 'AUTO'??
Better discuss with Yoni right away...